lib/types/socless_psuedo_states.ts

```diff
@@ -0,0 +1,47 @@
+import { BaseStateAction, PassState, StepFunction } from ".";
+
+// SOCless custom StepFunction (called a Playbook)
+export interface PlaybookDefinition extends StepFunction {
+  Playbook: string;
+  Decorators?: Record<string, unknown>;
+}
+
+// SOCless custom state (will be transformed by apb into an AWS state)
+export interface InteractionState extends BaseStateAction {
+  Type: "Interaction";
+  Resource: string;
+  TimeoutSeconds?: number;
+  TimeoutSecondsPath?: string;
+  HeartbeatSeconds?: number;
+  HeartbeatSecondsPath?: string;
+}
+
+// Deprecated?
+export interface HelperStateFinalized extends PassState {
+  Type: "Pass";
+  Result: {
+    Name: string;
+    Parameters: Record<string, any>;
+  };
+  ResultPath: "$.State_Config";
+  Next: string;
+}
+
+export interface SoclessTaskStepParameters {
+  "execution_id.$": "$.execution_id";
+  "artifacts.$": "$.artifacts";
+  "errors.$": "$.errors";
+  "results.$": "$.results";
+  State_Config: {
+    Name: string;
+    Parameters: Record<string, unknown>;
+  };
+}
+
+export interface SoclessInteractionStepParameters {
+  FunctionName: string; // AWS Lambda Resource
```

Comment:
From the serverless side, I believe this can also be a serverless reference i.e `${{self:custom.Integration}}`. Because when sls-apb first passes a playbook to `apb` for rendering, the serverless variables aren't resolved yet

Reply:
is the key still FunctionName?

Reply:
Yeah it is